product-service/serverless.ts

```diff
@@ -22,14 +21,48 @@ const serverlessConfiguration: AWS = {
     },
     environment: {
       AWS_NODEJS_CONNECTION_REUSE_ENABLED: '1',
+      SNS_ARN: { Ref: 'SNSTopic' },
     },
     lambdaHashingVersion: '20201221',
     httpApi: {
       cors: true,
     },
+    iamRoleStatements: [
+      {
+        Effect: 'Allow',
+        Action: ['sns:*'],
+        Resource: { Ref: 'SNSTopic' },
+      },
+    ],
+  },
+
+  resources: {
+    Resources: {
+      SNSTopic: {
+        Type: 'AWS::SNS::Topic',
+        Properties: { TopicName: 'import-service-topic' },
+      },
+      SNSSubscription: {
+        Type: 'AWS::SNS::Subscription',
+        Properties: {
+          Endpoint: 'zqn29865@zwoho.com',
+          Protocol: 'email',
+          TopicArn: { Ref: 'SNSTopic' },
+        },
+      },
+      SNSSubscriptionFilteredByTitle: {
+        Type: 'AWS::SNS::Subscription',
+        Properties: {
+          Endpoint: 'aqc45350@zwoho.com',
+          Protocol: 'email',
+          TopicArn: { Ref: 'SNSTopic' },
+          FilterPolicy: { event: ['product_added'], product_title: [{ prefix: 'iPhone' }] },
+        },
+      },
+    },
   },
   // import the function via paths
-  functions: { getProductsList, getProductsById, addProduct },
+  functions: { getProductsList, getProductsById, addProduct, catalogBatchProcess },
```

Comment:
As you've got index file in src/functions directory, you can import all functions as `import * as functions from '@functions/index';` and get rid of specifying each of them